### ISSUE TITLE
Bug with nested Scheduler

### ISSUE THREAD

**Issue by FrsECM** [huggingface/accelerate#3875]
### System Info

```Shell
- `Accelerate` version: 1.12.0
- Platform: Linux-6.8.0-85-generic-x86_64-with-glibc2.35
- `accelerate` bash location: /home/user/code/.venv/bin/accelerate
- Python version: 3.13.7
- Numpy version: 2.3.5
- PyTorch version: 2.9.1+cu128
- PyTorch accelerator: CUDA
- System RAM: 93.10 GB
- GPU type: NVIDIA RTX A6000
```

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [x] My own task or dataset (give details below)

### Reproduction

Hello ! 
I got an issue with accelerate when using nested schedulers. This can be usefull when we need to add a warmup for example.

Here is a code snippet to reproduce the issue :
```python
from torch.optim.lr_scheduler import LinearLR, CosineAnnealingLR, SequentialLR
import tqdm
import os
from dataclasses import dataclass,field
from omegaconf import OmegaConf
from bibcvmodels.utils import CheckpointManager,castfloat_or_ignore,grad_norm
from accelerate import Accelerator
from accelerate.utils import ProjectConfiguration
import torch
device = "cuda" if torch.cuda.is_available() else "cpu"

@dataclass
class TrainingConfig:
    lr:float = field(default=5e-4)
    # Training
    total_steps:int=field(default=100_000)
    warmup_steps:int=field(default=1_000)
    suffix:str="scheduler"

    
def main(cfg: TrainingConfig):
    torch.manual_seed(0)
    project_dir = CheckpointManager.create_project_dir('dev/outputs',suffix=cfg.suffix)
    os.makedirs('dev/outputs',exist_ok=True)
    # Create the accelerator
    project_config = ProjectConfiguration(
        project_dir=project_dir,
    )
    accelerator = Accelerator(
        project_config=project_config,
        mixed_precision='bf16',
        log_with="tensorboard",
    )
    accelerator.init_trackers('logs')

    # Create dummy models
    net = torch.nn.Identity()
    
    # Optimizer and scheduler
    cfg.lr=accelerator.num_processes*cfg.lr

    g1 = {"params": net.parameters(), "lr": cfg.lr, "weight_decay": 5e-4}
    opt = torch.optim.AdamW([g1])
    # We scale scheduler // accelerate
    warmup_steps_scheduler = cfg.warmup_steps*accelerator.num_processes
    total_steps_scheduler = cfg.total_steps**accelerator.num_processes
    s1 = LinearLR(opt, start_factor=0.01, total_iters=warmup_steps_scheduler)
    s2 = CosineAnnealingLR(opt, T_max=total_steps_scheduler - warmup_steps_scheduler, eta_min=1e-3*cfg.lr)
    scheduler = SequentialLR(opt, schedulers=[s1, s2], milestones=[warmup_steps_scheduler])

    # Register accelerate
    (
        net,
        opt,
        scheduler,
    ) = accelerator.prepare(
        net,
        opt,
        scheduler
    )

    current_step = 0
    # Training
    progress = tqdm.tqdm(range(cfg.total_steps))
    while current_step < cfg.total_steps:
        current_step += 1
        progress.update(1)
        opt.zero_grad()
        # accelerator.backward(loss)
        opt.step()
        scheduler.step()
        metrics = dict(lr_model=opt.param_groups[0]['lr'])
        progress.set_postfix(**castfloat_or_ignore(metrics))
        accelerator.log(castfloat_or_ignore(metrics,prefix='train/'),step=current_step)


if __name__ == "__main__":
    print('Read Omega Configuration...')
    conf = OmegaConf.merge(OmegaConf.structured(TrainingConfig()),OmegaConf.from_cli())
    print('Run Training...')
    main(conf)
```

The result differ when i use distributed or not.
Here are the tensorboard results of both cases :

<img width="1179" height="357" alt="Image" src="https://github.com/user-attachments/assets/0d3c1ac4-51f8-45cd-9cd4-7987c7332c18" />

- [x] The scaling is expected, because batch size is larger in distributed mode, i double the lr
- [ ]  **The Cosine Annealing is not triggered when distributed !**

What you can see on the orange curve is that the CosineAnnealing is not taken in account in the distributed case.
I ran the script this way : 
```bash
python dev/test_accelerate_distributed.py suffix="scheduler_one_process"
torchrun --nproc_per_node 2 dev/test_accelerate_distributed.py suffix="scheduler_two_process"
```
Do you have any idea about what's going on ?

### Expected behavior

I'd expect the annealing to work in distributed mode.

**Comment by FrsECM** [comment]
It was a bug from my side ;) Sorry for the disturbance.

I was doing :
- total_steps_scheduler = cfg.total_stepsaccelerator.num_processes
